python/src/greenbids/tailor/core/app/resources.py

```diff
@@ -69,32 +71,82 @@ def core_version(self) -> str:
     def is_ready(self) -> bool:
         return self._gb_model is not None
 
-    def refresh_model(self) -> None:
+    @property
+    def _dump_prefix(self):
+        return f"gb-tailor_{self.gb_model_name}_"
+
+    def _should_refresh(self) -> bool:
+        if self._gb_model is None:
+            return True
+
+        try:
+            rsp = requests.get(
+                settings.authenticated_index_url.geturl()
+                + f"/_commands/{settings.api_user}-{self.gb_model_name}.json"
+            )
```

Comment:
suggestion: timeout could be added to GET request to stop if the server fails to respond in after timeout time